### ISSUE TITLE
EMQ Websocket should only accept the 'mqtt' protocol

### ISSUE THREAD

**Issue by mworrell** [emqx/emqx#3914]
**Environment**:

- EMQ X version (e.g. `emqx_ctl status`): master

**What happened and what you expected to happen**:

1. If we connect to the Websocket controller with:

```javascript
self.socket = new WebSocket( self.remoteUrl, [ "mqtt.cotonic.org", "mqtt" ] );
```
I expect EMQ to respond with `mqtt` as the accepted protocol.

Instead it responds with 400 (due to the presence of multiple acceptable protocols)

2. If we connect to the Websocket controller with:

```javascript
let socket = new WebSocket( "ws://broker.emqx.io:8083/mqtt", [ "mqtt.cotonic.org" ] );
```
I expect EMQ to respond with an error, as the `mqtt.cotonic.org` protocol is not implemented by EMQ.

Instead it accepts `mqtt.cotonic.org`, where the EMQ controller can only handle `mqtt`

**How to reproduce it (as minimally and precisely as possible)**:

1. In a browser use:

```
new WebSocket( "ws://broker.emqx.io:8083/mqtt", [ "mqtt.cotonic.org", "mqtt" ] );
```

The EMQ controller will respond with 400.

2. In a browser use:

```
let socket = new WebSocket( "ws://broker.emqx.io:8083/mqtt", [ "mqtt.cotonic.org"] );
```

The accepted protocol in `socket.protocol` is `mqtt.cotonic.org`:

**Anything else we need to know?**:

In https://github.com/emqx/emqx/blob/master/src/emqx_ws_connection.erl#L190 the code needs to check for `<<"mqtt">>` in the list of possible protocols and not 1) accept only a single protocol and 2) accept any protocol starting with `mqtt`

**Comment by mworrell** [comment]
As reported in Cotonic https://github.com/cotonic/cotonic/issues/43

**Comment by tigercl** [comment]
@mworrell Thank you very much for your feedback, we will improve it as soon as possible.

**Comment by HJianBo** [comment]
Hi guys, these fixes had released v4.3-alpha.1
see: https://github.com/emqx/emqx/pull/4099

**Comment by mworrell** [comment]
Great work, thanks!
